### PR TITLE
corrected createColumnProvider

### PR DESCRIPTION
**Pull request by dsnyder-fairway**
issue was parentColumnIndex was always 0 when the columnCount was MAX_VALUE
If we had the columnCount as 1 we would end up with 4 columns. If we had columnCount as 3 we would have 2 columns.

### CODE REVIEW

File: webapp/src/main/webapp/src/app/aggregate-report/aggregate-report-summary.component.ts [hunk 1]
```diff
@@ -7,14 +7,18 @@ import { AssessmentDefinition } from './assessment/assessment-definition';
 import { Utils } from '../shared/support/support';
 import { SubgroupMapper } from './subgroup.mapper';
 
-const createColumnProvider = (columnCount: number = Number.MAX_VALUE): ColumnProvider => {
+const createColumnProvider = (columnCount: number = 20): ColumnProvider => {
```
Comment (dsnyder-fairway) [inline webapp/src/main/webapp/src/app/aggregate-report/aggregate-report-summary.component.ts]: defaulting to 20 as I feel that is a sufficiently large enough number of columns we should support

File: webapp/src/main/webapp/src/app/aggregate-report/aggregate-report-summary.component.ts [hunk 2]
```diff
@@ -7,14 +7,18 @@ import { AssessmentDefinition } from './assessment/assessment-definition';
...
   return (...sections) => {
     const parentColumns = [];
-    sections.forEach((section, index) => {
-      const parentColumnIndex = Math.floor(index / columnCount);
-      const columns = parentColumns[ parentColumnIndex ] = parentColumns[ parentColumnIndex ] || [];
-      columns.push(section);
-    });
+    const length = sections.length / columnCount;
+    for (let i = 0; i < sections.length; i++) {
```
Comment (malaffoon) [inline webapp/src/main/webapp/src/app/aggregate-report/aggregate-report-summary.component.ts]: I sorta see what's going on but i do have a knee-jerk whenever i see a non-forEach loop. Any way to do this that way?

Comment (dsnyder-fairway) [reply webapp/src/main/webapp/src/app/aggregate-report/aggregate-report-summary.component.ts]: I believe so. Let me see

File: webapp/src/main/webapp/src/app/aggregate-report/aggregate-report-summary.component.ts [hunk 3]
```diff
@@ -7,14 +7,18 @@ import { AssessmentDefinition } from './assessment/assessment-definition';
...
 import { SubgroupMapper } from './subgroup.mapper';
 
-const createColumnProvider = (columnCount: number = Number.MAX_VALUE): ColumnProvider => {
+const createColumnProvider = (columnCount: number = 20): ColumnProvider => {
   return (...sections) => {
     const parentColumns = [];
-    sections.forEach((section, index) => {
-      const parentColumnIndex = Math.floor(index / columnCount);
-      const columns = parentColumns[ parentColumnIndex ] = parentColumns[ parentColumnIndex ] || [];
-      columns.push(section);
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/aggregate-report/aggregate-report-summary.component.ts]: wondering why this didn't work - what was the breaking condition?

Comment (dsnyder-fairway) [reply webapp/src/main/webapp/src/app/aggregate-report/aggregate-report-summary.component.ts]: `const parentColumnIndex = Math.floor(index / columnCount);` often put parentColumnIdex = 0 (when large columnCounts were provided)

File: webapp/src/main/webapp/src/app/aggregate-report/aggregate-report-summary.component.ts (hunk 2 above)
Comment (esotrope) [reply webapp/src/main/webapp/src/app/aggregate-report/aggregate-report-summary.component.ts]: Could this all have been fixed by changing this:
```
const WideColumnProvider: ColumnProvider = createColumnProvider();
const WideColumnProvider: ColumnProvider = createColumnProvider(4);
```
?
I think my intent was to put a 4 there but forgot : /

Comment (dsnyder-fairway) [reply webapp/src/main/webapp/src/app/aggregate-report/aggregate-report-summary.component.ts]: `const WideColumnProvider: ColumnProvider = createColumnProvider(1);` gave us 4 columns

Comment (esotrope) [reply webapp/src/main/webapp/src/app/aggregate-report/aggregate-report-summary.component.ts]: gotcha - well whatever works is good for me right now 👍

**Comment by dsnyder-fairway** [comment]
meh, having a bit of trouble with the else statement here. Closing till I figure this out

**Pull request closed without merging**